packages/web-components/src/components/va-input-telephone/va-input-telephone.tsx

```diff
@@ -0,0 +1,247 @@
+ * @guidanceHref form/input-telephone
+ * @translations English
+ */
+
+@Component({
+  tag: 'va-input-telephone',
+  styleUrl: 'va-input-telephone.scss',
+  shadow: true,
+})
+export class VaInternationalTelephone {
+  @Element() el: HTMLElement;
+  textInputRef: HTMLVaTextInputElement;
+  DEFAULT_COUNTRY: CountryCode = 'US';
+
+  /**
+   * The telephone contact information
+   */
+  @Prop() contact?: string;
+
+  /**
+   * The 2 letter ISO country code for a country
+   */
+  @Prop({ reflect: true, mutable: true }) country?: CountryCode = this.DEFAULT_COUNTRY;
+
+
+  /**
+   * Header text for the component
+   */
+  @Prop() header?: string = 'Primary phone number';
+
+  /**
+   * Hint string text
+   */
+  @Prop() hint?: string;
+
+  /**
+   * The error for the component
+   */
+  @Prop({ reflect: true, mutable: true }) error?: string = '';
+
+  /**
+   * Whether the country select should be included. Set to true to exclude it.
+   */
+  @Prop() noCountry?: boolean = false;
+
+  /**
+   * The event emitted when the contact changes
+   */
+  @Event() vaContact: EventEmitter;
+
+  /**
+   * The event emitted when the country code changes
+   */
+  @Event() vaCountryCode: EventEmitter
+
+  /**
+   * The contact formatted according to the selected country
+   */
+  @State() formattedContact: string = '';
+
+  /**
+   * List of sorted countries used in the country combo-box
+   */
+  @State() countries: string[] = null;
+
+  /**
+   * Error for the country combo-box
+   */
+  @State() countryError: string = '';
+
+  /**
+   * Error for the contact input
+   */
+  @State() contactError: string = '';
+
+  resetErrors() {
+    this.countryError = '';
+    this.contactError = '';
+    this.error = '';
+  }
+
+  updateContact(event: InputEvent) {
+    const target = event.target as HTMLInputElement;
+    const { value } = target;
+    this.formattedContact = value ? this.formatContact(value) : '';
+  }
+
+  // get an error message for a country that includes the template of a valid phone number for that country
+  getErrorMessageForCountry() {
+    let example = getExampleNumber(this.country, examples).format('NATIONAL');
+    const asYouType = new AsYouType(this.country);
+    asYouType.input(example);
+    const msg = asYouType.getTemplate();
+    return `Enter a ${this.getCountryName(this.country)} phone number in a valid format, for example, [${msg}]`;
+  }
+
+  // validate the contact and emit the contact and validation state
+  validateContact() {
+    this.resetErrors();
+    let isValid = false;
+    if (this.country) {
+      if (this.formattedContact) {
+        isValid = isValidPhoneNumber(this.formattedContact, this.country);
+        this.error = isValid ? '' : this.getErrorMessageForCountry()
+      } else {
+        this.error = 'Please enter a phone number';
+      }
+      this.contactError = this.error;
+    } else {
+      this.validateCountry();
+    }
+    this.vaContact.emit({ contact: this.formattedContact, isValid });
```

Comment:
Question: This custom event is emitting regardless of if the number is valid or not. The event description does say `The event emitted when the country code changes` so I guess that tracks. But was it a specific request to receive both valid and invalid phone number entries?

I guess I'm just wondering if this could be simplified to only emitt when a number is valid but maybe you're following an existing onChange pattern that I'm forgetting about. Or you need that info internally somewhere.

Reply:
@jamigibbs I wondered this too. I went with always emitting the event and including an isValid flag because it does seem like we usually always emit an event on change.